coralproject/talk
# [CORL-2529] Always show new comment count when `NEW_COMMENT_COUNT` is enabled

## What does this PR do?

Some users want to be able to see whether they have visited a story and read the comments or not. By always showing the new comments since the user last visited the story (or never visited a story) we can indicated whether a story is new for them or if they have visited it in the past.

If they haven't ever seen a story and there are comments, the count script will show similar to: `3 Comments / 3 New`. This is because the story has never been visited and the 3 comments are both there and new.

If they have loaded a story and there are comments, the count script will show similar to: `22 Comments / 3 New`. This is because the story has been visited and since they last visited, 3 new comments were posted.

This helps commenters figure out whether they have visited a story or not and whether they should catch up on the commenting there.

## What changes to the GraphQL/Database Schema does this PR introduce?

No schema changes.

## Does this PR introduce any new environment variables or feature flags? 

No. Any new clients that want this feature will have to enable `NEW_COMMENT_COUNT`. Existing clients using this feature already have `NEW_COMMENT_COUNT` enabled and do not require any updates.

## If any indexes were added, were they added to `INDEXES.md`?

No new indexes.

## How do I test this PR?

* Build and run Coral
* Spin up `multi-site-test` or a similar host client that has count scripts hooked up
* See that the counts are showing, but no new counts are shown
* Enable `NEW_COMMENT_COUNT` with the following mutation:

    ```
     mutation {
       enableFeatureFlag(input: {
         flag: "NEW_COMMENT_COUNT",
         clientMutationId: "1"
      }) {
         flags
       }
     }
    ```
* Reload the page and see that new counts now show
* Visit articles and check that the new comment count always shows and is correct
 
## How do we deploy this PR?

No special considerations for deploy. Existing clients that use this already have the `NEW_COMMENT_COUNT` feature flag set.
